src/parser.rs

```diff
@@ -2233,53 +2233,166 @@ impl<'a> Parser<'a> {
     pub fn parse_copy(&mut self) -> Result<Statement, ParserError> {
         let table_name = self.parse_object_name()?;
         let columns = self.parse_parenthesized_column_list(Optional)?;
-        let to_or_from = self.expect_one_of_keywords(&[Keyword::FROM, Keyword::TO])?;
-        let to: bool = match to_or_from {
-            Keyword::TO => true,
-            Keyword::FROM => false,
-            _ => unreachable!("something wrong while parsing copy statment :("),
+        let to = match self.parse_one_of_keywords(&[Keyword::FROM, Keyword::TO]) {
+            Some(Keyword::FROM) => false,
+            Some(Keyword::TO) => true,
+            _ => self.expected("FROM or TO", self.peek_token())?,
         };
-        let mut filename = None;
-        // check whether data has to be copied form table or std in.
-        if !self.parse_keyword(Keyword::STDIN) {
-            filename = Some(self.parse_identifier()?)
-        }
-        // parse copy options.
-        let mut delimiter = None;
-        let mut csv_header = false;
-        loop {
-            if let Some(keyword) = self.parse_one_of_keywords(&[Keyword::DELIMITER, Keyword::CSV]) {
-                match keyword {
-                    Keyword::DELIMITER => {
-                        delimiter = Some(self.parse_identifier()?);
-                        continue;
-                    }
-                    Keyword::CSV => {
-                        self.expect_keyword(Keyword::HEADER)?;
-                        csv_header = true
-                    }
-                    _ => unreachable!("something wrong while parsing copy statment :("),
-                }
+        let target = if self.parse_keyword(Keyword::STDIN) {
+            CopyTarget::Stdin
+        } else if self.parse_keyword(Keyword::STDOUT) {
+            CopyTarget::Stdout
+        } else if self.parse_keyword(Keyword::PROGRAM) {
+            CopyTarget::Program {
+                command: self.parse_literal_string()?,
+            }
+        } else {
+            CopyTarget::File {
+                filename: self.parse_literal_string()?,
             }
-            break;
+        };
+        let _ = self.parse_keyword(Keyword::WITH);
```

Comment:
Shouldn't this be

```suggestion
        self.parse_keyword(Keyword::WITH)?;
```

So that the error is properly returned?

Reply:
The `WITH` keyword is optional, so the error should be ignored here.